network/robus/HAL/NATIVE/robus_hal.c

```diff
@@ -440,21 +442,27 @@ uint8_t RobusHAL_GetPTPState(uint8_t PTPNbr)
     {
         // Ask PTPA state to the server
         mg_ws_send(c, "PTPA1", sizeof("PTPA1"), WEBSOCKET_OP_TEXT);
-        // Wait for the server to acknowledge the PTP update
-        while (ptp_update == false)
+        // Wait for the server to acknowledge the PTP update if connected
+        if (ws_connected)
```

Comment:
If no broker exist it will timeout and you will be able to use localhost messaging only. if a broker appear during service execution, you will have to restart the service to restart WS connection.